src/core/WorkItemBuiltins.cpp

```diff
@@ -1229,7 +1229,11 @@ class WorkItemBuiltins
     int coordIndex = 1;
 
     // Check for sampler version
+#if LLVM_VERSION >= 140
+    if (callInst->arg_size() > 2)
+#else
     if (callInst->getNumArgOperands() > 2)
+#endif
```

Comment:
I couldn't find when `getNumArgOperands` was deprecated and replaced by `arg_size`, so not sure if this is actually necessary, but mostly followed other projects, e.g. https://github.com/JuliaLang/julia/pull/43722

Reply:
Looks like it was about [4 years ago](https://github.com/llvm/llvm-project/commit/05b5bd8b85288b2f5f56b5f61668813c46c44cda). The oldest LLVM version we support is 11 which is much newer than that, so I think we can just switch directly to the newer `arg_size` API instead of multi-versioning this code.

Reply:
Thank you for digging that up! I've removed the version check here and elsewhere.